### PR TITLE
Delete series and all child resources

### LINKED ISSUE #155: Delete dataset series from database
Add an endpoint to allow deleting a dataset series and everything beneath it (all releases, schemas, revisions, and appends) for users (i.e. Rob) to be able to keep a tidy list of datasets for demo purposes.

### PR DESCRIPTION
**Pull request by lkitching**
Fixes #155 - Add DELETE route for series which deletes the route along with all child resources (releases, revisions etc.). Also deletes any changes within the change store only referenced by the deleted series.

### CODE REVIEW

File: datahost-ld-openapi/deps.edn [hunk 1]
```diff
@@ -52,7 +52,9 @@
 
   :test {:extra-paths ["test" "env/test/resources"]
          :extra-deps {lambdaisland/kaocha {:mvn/version "1.85.1342"}
-                      lambdaisland/kaocha-junit-xml {:mvn/version "1.17.101"}}
+                      lambdaisland/kaocha-junit-xml {:mvn/version "1.17.101"}
+											com.gfredericks/test.chuck {:mvn/version "0.2.13"}
```
Comment (RickMoynihan) [inline datahost-ld-openapi/deps.edn:56]: I don't think this dependency is actually used here, can we remove it for now?

**Comment by RickMoynihan** [comment]
I've taken a look at this but whilst testing I discovered issue #260 which means that we can't strictly speaking remove everything.  It'd be good to fix that before merging this, either by fixing in  main and rebasing this ontop or by fixing here.

File: datahost-ld-openapi/deps.edn (hunk 1 above)
Comment (lkitching) [reply datahost-ld-openapi/deps.edn]: Thanks, removed!

**Review by rosado — COMMENTED** [review]
The CSVs uploaded by users are removed from the change store, but the store also holds the CSV snapshots (`:dh/revisionSnapshotCSV`), which are not cleaned up at the moment (or more precisely: `find-orphaned-changes` doesn't mention them).

**Comment by RickMoynihan** [comment]
Ahh ok, that explains it as I wasn't aware we'd implemented this.  

What I don't understand is why the two files have different hashes but are bitwise identical.

**Comment by RickMoynihan** [comment]
It seems that #265 which has now been merged with `main` fixes the issue I encountered on this branch with duplicate entries in the file-store. 

@lkitching can we rebase this PR onto main to get that fix?

**Comment by RickMoynihan** [comment]
Looks good to me!

**Pull request merged**